from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QApplication, QDialog, QPushButton, QMessageBox, QVBoxLayout
import mysql.connector as mc
from success import Ui_success

class Ui_new(object):

    def success(self):
         self.window = QtWidgets.QMainWindow()
         self.ui = Ui_success()
         self.title = "Success"
         self.ui.setupUi(self.window)
         self.window.show()

    def setupUi(self, new):
        new.setObjectName("new")
        new.resize(1042, 757)
        self.centralwidget = QtWidgets.QWidget(new)
        self.centralwidget.setObjectName("centralwidget")
        self.submit = QtWidgets.QPushButton(self.centralwidget)
        self.submit.setGeometry(QtCore.QRect(420, 600, 151, 71))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(18)
        self.submit.setFont(font)
        self.submit.setObjectName("submit")


        self.submit.clicked.connect(self.login)
        self.submit.clicked.connect(self.success)
        self.submit.clicked.connect(new.close)

        
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(0, 0, 1041, 91))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(48)
        self.label.setFont(font)
        self.label.setAlignment(QtCore.Qt.AlignCenter)
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(290, 130, 161, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_2.setFont(font)
        self.label_2.setAlignment(QtCore.Qt.AlignCenter)
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(self.centralwidget)
        self.label_3.setGeometry(QtCore.QRect(290, 210, 151, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_3.setFont(font)
        self.label_3.setAlignment(QtCore.Qt.AlignCenter)
        self.label_3.setObjectName("label_3")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(290, 370, 71, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_4.setFont(font)
        self.label_4.setAlignment(QtCore.Qt.AlignCenter)
        self.label_4.setObjectName("label_4")
        self.label_5 = QtWidgets.QLabel(self.centralwidget)
        self.label_5.setGeometry(QtCore.QRect(290, 330, 111, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_5.setFont(font)
        self.label_5.setAlignment(QtCore.Qt.AlignCenter)
        self.label_5.setObjectName("label_5")
        self.label_6 = QtWidgets.QLabel(self.centralwidget)
        self.label_6.setGeometry(QtCore.QRect(290, 290, 221, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_6.setFont(font)
        self.label_6.setAlignment(QtCore.Qt.AlignCenter)
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.centralwidget)
        self.label_7.setGeometry(QtCore.QRect(290, 250, 171, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_7.setFont(font)
        self.label_7.setAlignment(QtCore.Qt.AlignCenter)
        self.label_7.setObjectName("label_7")
        self.label_8 = QtWidgets.QLabel(self.centralwidget)
        self.label_8.setGeometry(QtCore.QRect(290, 490, 191, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_8.setFont(font)
        self.label_8.setAlignment(QtCore.Qt.AlignCenter)
        self.label_8.setObjectName("label_8")
        self.label_9 = QtWidgets.QLabel(self.centralwidget)
        self.label_9.setGeometry(QtCore.QRect(290, 450, 211, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_9.setFont(font)
        self.label_9.setAlignment(QtCore.Qt.AlignCenter)
        self.label_9.setObjectName("label_9")
        self.label_10 = QtWidgets.QLabel(self.centralwidget)
        self.label_10.setGeometry(QtCore.QRect(290, 410, 171, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_10.setFont(font)
        self.label_10.setAlignment(QtCore.Qt.AlignCenter)
        self.label_10.setObjectName("label_10")
        self.label_11 = QtWidgets.QLabel(self.centralwidget)
        self.label_11.setGeometry(QtCore.QRect(290, 530, 81, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_11.setFont(font)
        self.label_11.setAlignment(QtCore.Qt.AlignCenter)
        self.label_11.setObjectName("label_11")
        self.pname = QtWidgets.QLineEdit(self.centralwidget)
        self.pname.setGeometry(QtCore.QRect(450, 130, 271, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.pname.setFont(font)
        self.pname.setObjectName("pname")
        self.blood = QtWidgets.QLineEdit(self.centralwidget)
        self.blood.setGeometry(QtCore.QRect(440, 210, 281, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.blood.setFont(font)
        self.blood.setObjectName("blood")
        self.disease = QtWidgets.QLineEdit(self.centralwidget)
        self.disease.setGeometry(QtCore.QRect(460, 250, 261, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.disease.setFont(font)
        self.disease.setObjectName("disease")
        self.haemoglobin = QtWidgets.QLineEdit(self.centralwidget)
        self.haemoglobin.setGeometry(QtCore.QRect(510, 290, 211, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.haemoglobin.setFont(font)
        self.haemoglobin.setObjectName("haemoglobin")
        self.allergies = QtWidgets.QLineEdit(self.centralwidget)
        self.allergies.setGeometry(QtCore.QRect(400, 330, 321, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.allergies.setFont(font)
        self.allergies.setObjectName("allergies")
        self.fees = QtWidgets.QLineEdit(self.centralwidget)
        self.fees.setGeometry(QtCore.QRect(360, 370, 361, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.fees.setFont(font)
        self.fees.setObjectName("fees")
        self.gardname = QtWidgets.QLineEdit(self.centralwidget)
        self.gardname.setGeometry(QtCore.QRect(460, 410, 261, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.gardname.setFont(font)
        self.gardname.setObjectName("gardname")
        self.gardphone = QtWidgets.QLineEdit(self.centralwidget)
        self.gardphone.setGeometry(QtCore.QRect(500, 450, 221, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.gardphone.setFont(font)
        self.gardphone.setObjectName("gardphone")
        self.add = QtWidgets.QLineEdit(self.centralwidget)
        self.add.setGeometry(QtCore.QRect(480, 490, 241, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.add.setFont(font)
        self.add.setObjectName("add")
        self.label_12 = QtWidgets.QLabel(self.centralwidget)
        self.label_12.setGeometry(QtCore.QRect(270, 170, 151, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.label_12.setFont(font)
        self.label_12.setAlignment(QtCore.Qt.AlignCenter)
        self.label_12.setObjectName("label_12")
        self.pid = QtWidgets.QLineEdit(self.centralwidget)
        self.pid.setGeometry(QtCore.QRect(410, 170, 311, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.pid.setFont(font)
        self.pid.setObjectName("pid")
        self.notes = QtWidgets.QLineEdit(self.centralwidget)
        self.notes.setGeometry(QtCore.QRect(370, 540, 351, 31))
        font = QtGui.QFont()
        font.setFamily("Maiandra GD")
        font.setPointSize(14)
        self.notes.setFont(font)
        self.notes.setObjectName("notes")
        new.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(new)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1042, 26))
        self.menubar.setObjectName("menubar")
        new.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(new)
        self.statusbar.setObjectName("statusbar")
        new.setStatusBar(self.statusbar)

        self.retranslateUi(new)
        QtCore.QMetaObject.connectSlotsByName(new)

    def retranslateUi(self, new):
        _translate = QtCore.QCoreApplication.translate
        new.setWindowTitle(_translate("new", "new"))
        self.submit.setText(_translate("new", "Submit"))
        self.label.setText(_translate("new", "HOSPITAL MANAGEMENT"))
        self.label_2.setText(_translate("new", "Patient Name:"))
        self.label_3.setText(_translate("new", "Blood Group:"))
        self.label_4.setText(_translate("new", "Fees:"))
        self.label_5.setText(_translate("new", "Allergies:"))
        self.label_6.setText(_translate("new", "Haemoglobin Leves:"))
        self.label_7.setText(_translate("new", "Disease Caused:"))
        self.label_8.setText(_translate("new", "Patient\'s Address:"))
        self.label_9.setText(_translate("new", "Gardian Phone No:"))
        self.label_10.setText(_translate("new", "Gardian Name:"))
        self.label_11.setText(_translate("new", "Notes:"))
        self.label_12.setText(_translate("new", "Patient ID"))


    def login(self):

        
        try:
            pname = self.pname.text()
            pid = self.pid.text()
            blood = self.blood.text()
            disease = self.disease.text()
            haemoglobin = self.haemoglobin.text()
            allergies = self.allergies.text()
            fees = self.fees.text()
            gardname = self.gardname.text()
            gardphone = self.gardphone.text()
            patiendadd = self.add.text()
            notes = self.notes.text()
            
            mydb = mc.connect(
                host = 'localhost',
                user='root',
                password='password',
                database='ohms'

            )

            mycursor = mydb.cursor()
            query = "insert into overflow values('" + pname + "','"+ pid + "','" + blood + "','" + disease + "'," + haemoglobin + ",'" + allergies + "'," + fees + ",'" + gardname + "'," + gardphone + ",'" + patiendadd + "','" + notes + "');"
            mycursor.execute(query)
            mydb.commit()
            mydb.close()
            var = True
            print(query)
            print("Successful")

            

        except mc.Error as e:
            Var = False
            print("smth is wrong i can feel it")
            print(query)
            


    

if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    new = QtWidgets.QMainWindow()
    ui = Ui_new()
    ui.setupUi(new)
    new.show()
    sys.exit(app.exec_())
